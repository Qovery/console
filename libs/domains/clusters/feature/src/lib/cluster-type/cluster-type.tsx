import { type CloudProviderEnum, KubernetesEnum } from 'qovery-typescript-axios'
import { P, match } from 'ts-pattern'
import { Badge, type BadgeProps } from '@qovery/shared/ui'

export interface ClusterTypeProps extends Omit<BadgeProps, 'color'> {
  cloudProvider: keyof typeof CloudProviderEnum
  kubernetes?: KubernetesEnum
}

export function ClusterType({ cloudProvider, kubernetes, ...props }: ClusterTypeProps) {
  const clusterType = match([cloudProvider, kubernetes])
    .with(['AWS', KubernetesEnum.K3_S], () => 'EC2 (K3S)')
    .with(['AWS', KubernetesEnum.MANAGED], ['AWS', undefined], () => 'EKS')
    .with(['AWS', KubernetesEnum.SELF_MANAGED], ['AWS', undefined], () => 'Self-managed')
    // Scaleway
    .with(['SCW', P._], () => 'Kapsule')
    // Google GCP
    .with(['GCP', P._], () => 'GKE (Autopilot)')
    // BYOK
    .with(['ON_PREMISE', P._], () => 'On-premise')
    .exhaustive()
  return (
    <Badge color="neutral" {...props}>
      {clusterType}
    </Badge>
  )
}

export default ClusterType
