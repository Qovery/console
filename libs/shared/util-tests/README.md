# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
