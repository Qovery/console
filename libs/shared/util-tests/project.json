{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
