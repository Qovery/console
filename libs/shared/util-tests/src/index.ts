export * from './lib/shared-util-tests'
