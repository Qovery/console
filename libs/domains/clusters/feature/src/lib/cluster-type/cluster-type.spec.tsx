import { KubernetesEnum } from 'qovery-typescript-axios'
import { renderWithProviders } from '@qovery/shared/util-tests'
import { ClusterType } from './cluster-type'

describe('ClusterType', () => {
  it('should render as K3S', () => {
    const { baseElement } = renderWithProviders(<ClusterType cloudProvider="AWS" kubernetes={KubernetesEnum.K3_S} />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should render as EKS', () => {
    const { baseElement } = renderWithProviders(<ClusterType cloudProvider="AWS" kubernetes={KubernetesEnum.MANAGED} />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should render as SCW', () => {
    const { baseElement } = renderWithProviders(<ClusterType cloudProvider="SCW" />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should render as GCP', () => {
    const { baseElement } = renderWithProviders(<ClusterType cloudProvider="GCP" />)
    expect(baseElement).toMatchSnapshot()
  })
})
