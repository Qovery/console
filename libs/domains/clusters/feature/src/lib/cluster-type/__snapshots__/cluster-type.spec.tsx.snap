// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterType should render as EKS 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded border-neutral-250 text-neutral-350 dark:border-neutral-350 dark:text-neutral-300"
      data-accent-color="neutral"
    >
      EKS
    </span>
  </div>
</body>
`;

exports[`ClusterType should render as GCP 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded border-neutral-250 text-neutral-350 dark:border-neutral-350 dark:text-neutral-300"
      data-accent-color="neutral"
    >
      GKE (Autopilot)
    </span>
  </div>
</body>
`;

exports[`ClusterType should render as K3S 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded border-neutral-250 text-neutral-350 dark:border-neutral-350 dark:text-neutral-300"
      data-accent-color="neutral"
    >
      EC2 (K3S)
    </span>
  </div>
</body>
`;

exports[`ClusterType should render as SCW 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded border-neutral-250 text-neutral-350 dark:border-neutral-350 dark:text-neutral-300"
      data-accent-color="neutral"
    >
      Kapsule
    </span>
  </div>
</body>
`;
